src/SelfScreener/GeneralSymptoms.spec.tsx

```diff
@@ -0,0 +1,58 @@
+import React from "react"
+import { fireEvent, render } from "@testing-library/react-native"
+import { useNavigation } from "@react-navigation/native"
+
+import { SelfScreenerContext } from "../SelfScreenerContext"
+import GeneralSymptoms from "./GeneralSymptoms"
+import { GeneralSymptom } from "./selfScreener"
+import { factories } from "../factories"
+import { SelfScreenerStackScreens } from "../navigation"
+
+jest.mock("@react-navigation/native")
+describe("GeneralSymptoms", () => {
+  describe("updating symptoms", () => {
+    it("calls the update function", () => {
+      expect.assertions(1)
+      const updateSymptomsSpy = jest.fn()
+      const context = factories.selfScreenerContext.build({
+        updateGeneralSymptoms: updateSymptomsSpy,
+      })
+
+      const { getByText } = render(
+        <SelfScreenerContext.Provider value={context}>
+          <GeneralSymptoms />
+        </SelfScreenerContext.Provider>,
+      )
+
+      const achingCheckbox = getByText("self_screener.general_symptoms.aching")
+
+      fireEvent.press(achingCheckbox)
+      expect(updateSymptomsSpy).toHaveBeenCalledWith(GeneralSymptom.ACHING)
+    })
+  })
+
+  describe("pressing next", () => {
+    it("navigates to the summary screen", () => {
+      const navigationSpy = jest.fn()
+      ;(useNavigation as jest.Mock).mockReturnValue({
+        navigate: navigationSpy,
+      })
+      const context = factories.selfScreenerContext.build({
+        generalSymptoms: [GeneralSymptom.COUGH],
```

Comment:
Is having selected at least one symptom a pre-requisite to allow the user to move to the next screen? If such should we express that with the `it` description or perhaps another example?

Reply:
This is a good question, and I'm not confident at the answer right now! The mocks mention being able to "skip" questions, but that's not in the design yet. So I'm not sure I want to encode that as a desired behavior in a test just yet 🤔